### DISCUSSION TITLE
How to get cable force when actuated with displacement?

### DISCUSSION THREAD

**Discussion by qpding** [SofaDefrost/SoftRobots#141] [Help - Programming with SoftRobots] [answered]
### TL;DR
Is there a way to get the cable force (tension) when _**valueType="displacement"**_ in a _**CableConstraint**_ object?

### Description
I'm doing a simulation on a cable-driven elastic object. Defined the elastic object with _**ElasticMaterialObject**_ from stlib3.physics.deformable, and the cable using
```
from softrobots.actuators import PullingCable
...
cable_t = PullingCable(eobject,
        name="PullingCableTop",
        cableGeometry=loadPointListFromFile(cable_point_list),
        pullPointLocation=[length, y, 0.0],
        translation=[0.0, y, 0.0],
        rotation=[0.0, 0.0, 0.0])
```
Read the doc from [here](https://softrobots.readthedocs.io/en/latest/_modules/softrobots/actuators/cable.html?highlight=CableConstraint#) and figured out that either force or displacement can be solved. I'm wondering if there exists a way to actuate the elastomer with cable displacement, and get the solved cable tension force at the same time, sort of like a force sensor?

### Solution?
Tried this and got output:
```
print(self.cable_t.CableConstraint.force.value)
```
Can someone confirm if this is correct?

Quinn

**Comment by EulalieCoevoet** [comment, marked as answer]
Yes it is correct. You can access the read only data fields `force` and `displacement` to get the actual/current value of both, either your input is a force or displacement. 

![image](https://user-images.githubusercontent.com/12150715/167114777-6646c23e-9a91-4945-b5bb-b2e85d57b710.png)
